ur_robot_driver/README.md

```diff
@@ -4,6 +4,74 @@ This package contains the actual driver for UR robots. It is part of the *univer
 
+
+## About this branch
+
+This branch is a beta test of new interfaces towards the robot.
+It is, as such, to be considered under development, and documentation may in parts be missing and
+things might change in the near future.
+It is required to use the beta-testing branch of the [Universal Robots Client Library](https://github.com/UniversalRobots/Universal_Robots_Client_Library). 
+
+Additionally, the packages in the [cartesian ros control
+repository](https://github.com/fzi-forschungszentrum-informatik/cartesian_ros_control) are needed,
+both for the provided cartesian interface definitions and new controllers utilizing the new
+interfaces to the robot.
+
+### Features
+
+This branch introduces several new interfaces via which to communicate with the robot.
+
+* Cartesian pose streaming
+* Cartesian velocity streaming
+* Joint-based trajectory forwarding
+* Cartesian trajectory forwarding
+* Cartesian control using the joint-based robot interface
+
+All of them can be used with corresponding ROS controllers.
+More information about the controllers can be found in the cartesian ros control repository.
+
+##### Cartesian pose streaming
+
+Start the preloaded `pose_traj_controller`.
+This controller allows ROS-level interpolation and streaming of cartesian poses to the robot.
+It uses the newly created FollowCartesianTrajectory action.
+
+##### Cartesian velocity streaming
+
+Start the preloaded `twist_controller`.
+This controller allows for streaming twist commands via the `twist_controller/command` topic to be
+interpreted as cartesian velocity control commands by the robot.
+
+##### Joint-based trajectory forwarding
+
+Start the preloaded `forward_joint_traj_controller`.
+This controllers allows for forwarding full trajectories to the robot, to be interpolated and
+executed completely by the internal robot control.
+While it uses the same FollowJointTrajectory interface as the previously existing trajectory
+control, during execution no control commands other than a keepalive signal are sent to the robot.
+A cancellation of the trajectory is still possible.
+
+Of special not for the forwarded trajectory is that, currently, blending is activated on the robot.
+This produces smoother trajectories, but causes target points other than the last to not
+necessarily be reached.
+One of the objectives of this beta test is to gather feedback for this behaviour and look into
+feasible ways to control this blending radius, that has to directly corresponding field in the ROS
+trajectory interface definition, to be set based on the commanded trajectory.
+
+##### Cartesian trajectory forwarding
+
+Start the preloaded `forward_cartesian_traj_controller`.
+Similar to the previous controller, this offers forwarding cartesian trajectories to the robot.
+It is used in the same general structure and also utilizes blending.
+
+##### Cartesian control using joint-based robot interface
+
+Start the preloaded `vendor_cartesian_traj_controller`.
```

Comment:
What's the rationale for prefixing this with `vendor_`?

Before reading the explanatory text below, I sort-of assumed this would be a pure forwarding controller, but it seems that's the role of the `forward_cartesian_traj_controller`.

Reply:
`vendor_ ` is an artifact from a different context. I would suggest dropping this and use the more verbose names:

`joint_based_cartesian_traj_controller` and `pose_based_cartesian_traj_controller`.